tests/services/test_file_mover.py

```
@@ -240,19 +247,37 @@ def test_move_with_rollback_success(mock_move, mock_verifier, source_and_dest):
+    dest.touch()
+
+    # Patch Path.exists and Path.unlink at the class level
+    mock_path_exists = mocker.patch("pathlib.Path.exists", autospec=True)
+    mock_path_unlink = mocker.patch("pathlib.Path.unlink", autospec=True)
+
+    # Store the original Path.exists method before patching
+    _original_path_exists = Path.exists
+
+    # Configure mock_path_exists to return True for the 'dest' object in rollback check
+    # For any other path, call the *original* Path.exists method using the stored reference
+    def custom_exists_side_effect(path_obj):
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `custom_exists_side_effect`

(ANN202)